pkg/artifact/job.go

```diff
@@ -0,0 +1,83 @@
+package artifact
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"github.com/abcxyz/github-metrics-aggregator/pkg/version"
+	"github.com/abcxyz/pkg/logging"
+	"github.com/abcxyz/pkg/workerpool"
+)
+
+// ExecuteJob runs the ingestion pipeline job to read GitHub
+// action workflow logs from GitHub and store them into GCS.
+func ExecuteJob(ctx context.Context, cfg *Config) error {
+	logger := logging.FromContext(ctx)
+
+	bq, err := NewBigQuery(ctx, cfg.ProjectID, cfg.DatasetID)
+	if err != nil {
+		return fmt.Errorf("failed to create bigquery client: %w", err)
+	}
+	defer bq.Close()
+
+	eventsTableDotNotation := formatGoogleSQL(cfg.ProjectID, cfg.DatasetID, cfg.EventsTableID)
+	artifactsTableDotNotation := formatGoogleSQL(cfg.ProjectID, cfg.DatasetID, cfg.ArtifactsTableID)
+
+	// Create a pool of workers to manage all of the log ingestions
+	pool := workerpool.New[ArtifactRecord](&workerpool.Config{
+		Concurrency: int64(runtime.NumCPU()),
+		StopOnError: false,
+	})
+
+	// Setup a log ingester to process ingestion events
+	logsFn, err := NewLogIngester(ctx, cfg.BucketName, cfg.GitHubAppID, cfg.GitHubInstallID, cfg.GitHubPrivateKey)
+	if err != nil {
+		return fmt.Errorf("failed to create log ingester: %w", err)
+	}
+
+	logger.InfoContext(ctx, "ingestion job starting",
+		"name", version.Name,
+		"commit", version.Commit,
+		"version", version.Version)
+
+	// Read up to `BatchSize` number of events that need to be processed
+	query := fmt.Sprintf(SourceQuery, eventsTableDotNotation, artifactsTableDotNotation, cfg.BatchSize)
```

Comment:
Is this vulnerable to sql injection or is there some prepared statement like functionality I am missing?

Reply:
This doesn't accept any user input, the input is all from the environment. I guess in theory a malicious environment variable could be set but if you are able to update that then you have much easier attack vectors than trying to make the cloud run job do crazy things imo. AFAIK you can't set the table ids as parameters so the only thing that I could turn in to a query parameter would be the batch size which is an int and not a string.

Reply:
ack